recipes/server_check.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_relative './server_check_helper'
+
+consul_service_file = '/etc/systemd/system/consul.service'
+consul_service_folder = '/etc/systemd/system/consul.service.d'
+drop_in_file = "#{consul_service_folder}/server_start_prevention.conf" # prevent systemd from starting
+
+directory consul_service_folder do
+  owner 'root'
+  mode '0744'
+  action :create
+end
+
+# Output file is used for systemd to check whether it should start consul.
+if is_server_disallowed?(node['consul']['config']['retry_join'], node['ipaddress'])
+  # Stop the service if it is already running.
+  service 'consul' do
+    action :stop
+    only_if { ::File.exist?(consul_service_file) }
+  end
+
+  # Update config itself to further prevent it from becoming a server.
+  node.override['consul']['config']['server'] = false
+
+  # Currently has to use ConditionPathExists instead of ExecCondition.
+  # Blocks the service from starting if this file itself is present.
+  # ExecCondition requires systemd 243+, Ubuntu 1804 has systemd 237.
+  file drop_in_file do
+    owner 'root'
+    mode '0744'
+    content <<-EOF
+        [Unit]
+        ConditionPathExists=!#{drop_in_file}
+        EOF
+    action :create
+  end
+else
+  # For non rogue servers, simply remove the drop-in file to unblock service from starting.
+  file drop_in_file do
+    action :delete
+  end
+end
+
+# This is required after changing the systemd unit file.
+bash 'reload systemd daemon' do
```

Comment:
Will this happen every time, even with no changes? Do we want that?

Reply:
This will make systemd check for new configuration files every time, there is no impact for already running services per documentation. Currently I am just relying Chef resource's idempotency property for creating/deleting the conf file, but I can add an explicit file check to only load configuration if there is a change.

Reply:
Fixed in https://github.com/Roblox/consul-cookbook/pull/44/commits/a3210fa5d0d7047e244e6446f47c90bc1ccd1a90